magpie/models.py

```diff
@@ -1042,13 +1040,51 @@ class NetworkToken(BaseModel, Base):
 
     def __init__(self, *_, **__):
         super(NetworkToken, self).__init__(*_, **__)
-        if not self.token:
-            self.token = uuid.uuid4()
+        if self.token:
+            # The token should not be created with the object, instead call refresh_token after the object is created
+            # Otherwise, the unhashed token will never be available to the user
+            self.token = None
 
-    token = sa.Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4, unique=True)
-    user_id = sa.Column(sa.Integer,
-                        sa.ForeignKey("users.id", onupdate="CASCADE", ondelete="CASCADE"), unique=True)
-    user = relationship("User", foreign_keys=[user_id])
+    @staticmethod
+    def _hash_token(token):
+        # type: (Str) -> Str
+        if isinstance(token, str):
+            token = uuid.UUID(token)
+        h = hashlib.sha256()
+        h.update(token.bytes)
+        return h.hexdigest()
+
+    def refresh_token(self):
+        # type: () -> str
+        unhashed_token = str(uuid.uuid4())
+        self.token = self._hash_token(unhashed_token)
+        self.created = datetime.datetime.utcnow()
+        return unhashed_token
+
+    def expired(self):
+        # type: () -> bool
+        expiry = int(get_constant("MAGPIE_NETWORK_DEFAULT_TOKEN_EXPIRY"))
+        return (datetime.datetime.utcnow() - self.created) > expiry
+
+    @classmethod
+    def get_expired(cls, db_session):
```

Comment:
Unless I didn't see properly, the only use is in the CLI to call `delete()`. Would be better to simply have `delete_expired` instead to avoid possible mishandling by the CLI.